streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -62,92 +62,59 @@ public interface KStream<K, V> {
     /**
      * Create a new {@code KStream} that consists of all records of this stream which satisfy the given predicate.
      * All records that do not satisfy the predicate are dropped.
-     * This is a stateless record-by-record operation.
+     * This is a stateless record-by-record operation (cf. {@link #processValues(FixedKeyProcessorSupplier, String...)}
+     * for stateful record processing).
+     *
+     * @param predicate
+     *        a filter {@link Predicate} that is applied to each record
+     *
+     * @return A {@code KStream} that contains only those records that satisfy the given predicate.
```

Comment:
super-nit: I don't necessarily follow this myself (I typically have `@param` and `@return` both start with a capital and end with a period), but the official recommendation is to not capitalize and not follow with a period, as it was before ( https://www.oracle.com/de/technical-resources/articles/java/javadoc-tool.html ).

Also, fun fact, there are people on github called `@param` and `@return` that must be pinged all day in reviews.

Reply:
![Screenshot 2025-01-30 at 9 25 18 AM](https://github.com/user-attachments/assets/4c9760a7-4477-4348-b238-11e6de9c49bb)

Well, looking at how JavaDocs are actually rendered, it seems to be better to use capitalization and period. After `Returns:` there is a new line -- it's weird if it start with lower case IMHO. Also, we sometime use more than one sentence to describe what is returned, and it make it quite weird if we don't use capitalization and periods if it's two sentences.

For (method and type/generic) parameters, given the rendering as single line as `<name> - <description>` with a dash in between, I think the recommendation does make sense though (even if using capitalization and periods at the end might still be ok.)

At least that's why I did it the way I did it. 🤷